nodejs/webapi/server.js

```diff
@@ -0,0 +1,93 @@
+const express = require("express"); // import express
+const { swaggerSpecs, setupSwaggerUi } = require("./swagger/openapiservice");
+
+// JSDoc comments for api endpoints
+/**
+ * @swagger
+ * /weatherforecast:
+ *   get:
+ *    description: Use to request weather data
+ *    operationId: getWeatherData
+ *    tags:
+ *    - Weather
+ *    responses:
+ *     200:
+ *      description: A successful response
+ *      content:
+ *       application/json:
+ *        schema:
+ *         type: array
+ *         items:
+ *          type: object
+ *          properties:
+ *           date:
+ *            type: string
+ *            format: date
+ *            description: Date of the weather forecast
+ *            example: '2024-06-05'
+ *           temperatureC:
+ *            type: integer
+ *            description: Temperature in Celsius
+ *            example: 51
+ *           summary:
+ *            type: string
+ *            description: Weather summary
+ *            example: Cool
+ *           temperatureF:
+ *            type: integer
+ *            description: Temperature in Fahrenheit
+ *            example: 123
+ */
+
+const app = express();
+const port = process.env.PORT || 3030;
+
+// Random weather data
+const generateRandomTemperatureC = () => Math.floor(Math.random() * 61) - 20;
+const generateRandomSummary = () =>
+  ["Hot", "Warm", "Mild", "Cool", "Bracing", "Scorching", "Freezing"][
+    Math.floor(Math.random() * 7)
+  ];
+
+// Function to generate random weather data for a specified number of days
+const generateRandomWeatherData = (numDays) => {
+  const startDate = new Date();
+  return Array.from({ length: numDays }, (_, i) => {
+    const date = new Date(startDate); // Create a new date instance for each entry
+    date.setDate(startDate.getDate() + i);
+    const temperatureC = generateRandomTemperatureC();
+    const temperatureF = Math.round((temperatureC * 9) / 5 + 32);
+    const summary = generateRandomSummary();
+    return {
+      date: date.toISOString().split("T")[0],
+      temperatureC,
+      summary,
+      temperatureF,
+    };
+  });
+};
+
+// send weather data to /weatherdata
+app.get("/weatherforecast", (req, res) => {
+  // Generate random weather data for the next 5 days
+  const weatherData = generateRandomWeatherData(5);
+  res.json(weatherData);
+});
+
+// route to serve basic/ improved JSON, Set up Swagger UI and redirect /
+app.get("/api-docs/swagger.json", (req, res) => {
+  res.setHeader("Content-Type", "application/json");
+  res.send(swaggerSpecs.improved);
```

Comment:
Please comment out this line, which causes the error in the first place like:

![image](https://github.com/Azure-Samples/APICenter-Reference/assets/1538528/d39fb614-10b3-48d4-85ab-f1cd4d970a98)